qutip/tests/test_utilities.py

```diff
@@ -105,3 +105,9 @@ def test_unit_clebsch_delta_m(j1, j2):
                 sum_differ += c1*c2
         assert sum_match == pytest.approx(1)
         assert sum_differ == pytest.approx(int(m1 == m1p and m2 == m2p))
+
+
+def test_cpu_count():
+    ncpus = available_cpu_count()
```

Comment:
It would be good to test more of the cases if we can:

* We could test the environment variable cases using pytest's `monkeypatch.setenv`.
* I guess `os` vs `multiprocessing` we can test by letting them run on different enviroments one day (probably the `os` version doesn't work on Windows?)

Maybe just testing the environment variable overrides is enough for now.

Reply:
I added a test with `monkeypatch`. Yes it would be better to test more situations, but I think we will need to test them manually.

I looked deeper and both `os` and `psutil` do not add the affinity function when unavailable on the os. So I rewrote the function with a less `try`.